lib/api/listParts.js

```diff
@@ -167,9 +210,16 @@ function listParts(authInfo, request, log, callback) {
                 // - {uploadId}
                 // - {splitter}
                 // - {partNumber}
-                const index = item.key.lastIndexOf(splitter);
+                let partNumber;
+                if (item.key) {
+                    const index = item.key.lastIndexOf(splitter);
+                    partNumber =
+                        parseInt(item.key.substring(index + splitterLen), 10);
+                } else {
```

Comment:
please add note that otherwise the listing came from real AWS so has partNumber as attribute already